HDOS-703

pka1024 opened the pull request.

Commit:
feat(angular 14 update):
- ng update @angular/core@14 @angular/cli@14 --force.

Commit:
feat(angular 14 update):
- ng update @angular/material@14 --force.

Commit:
feat(angular 14 update):
- ng update @ngrx/store@14.

Commit:
- npm i @danielmoncada/angular-datetime-picker@14.2.0.
- npm i @danielmoncada/angular-datetime-picker-moment-adapter@3.0.1.

Commit:
feat(angular 14 update):
- npm update.

Commit:
feat(angular 14 update):
- Fixed deprecated warning  "~" import in scss.
- Fixed katex no known default import.
- Fixed Angular material expansion-panel.

Commit:
feat(angular 14 update):
- Prettier style changes.
- Fixed circular dependency error, after upgrading "angular-auth-oidc-client".
- Fixed config loading and getting async.

Commit:
feat(angular 14 update):
- Fixed a compatibly issue with
"prettier" >= v3.0.0. It needed "eslint-plugin-prettier" >= v5.0.0 and "eslint-plugin-jsdoc" < v43.0.0 to work or
else linting throws errors in node_modules.

pka1024 committed:
angular 14 update

feat(angular 14 update):
- Updated eslint.
- Fixed unit-tests errors.
- Finished angular-auth-oidc-client migration.

pka1024 committed:
angular 14 update

feat(angular 14 update):
- Switched from "ngx-monaco-editor" to "ngx-monaco-editor-v2" because it's support stopped with angular-v12.

pka1024 committed:
angular 14 update

feat(angular 14 update):
- Switched from "ngx-monaco-editor" to "ngx-monaco-editor-v2" because it's support stopped with angular-v12.

pka1024 committed:
Merge branch 'hetida:release' into feature/angular-update-14

pka1024 committed:
angular 15 update

feat(angular 15 update):
- Updated typescript.

pka1024 committed:
angular 15 update

feat(angular 15 update):
- ng update @angular/core@15 @angular/cli@15 --force.

pka1024 committed:
angular 15 update

feat(angular 15 update):
- ng update @angular/material@15 --force.

pka1024 committed:
angular 15 update

angular 15 update
- ng update @ngrx/store@15.

pka1024 committed:
angular 15 update

feat(angular 15 update):
- npm i angular-auth-oidc-client@15.0.5.

pka1024 committed:
angular 15 update

feat(angular 15 update):
- npm i @danielmoncada/angular-datetime-picker@15.0.2.
- npm i ngx-monaco-editor-v2@15.0.1.
- npm i monaco-editor@0.34.1.

pka1024 committed:
angular 15 update

feat(angular 15 update):
- npm i @angular-eslint/builder@15.2.1.
- npm i @angular-eslint/eslint-plugin@15.2.1.
- npm i @angular-eslint/eslint-plugin-template@15.2.1.
- npm i @angular-eslint/schematics@15.2.1.
- npm i @angular-eslint/template-parser@15.2.1.

pka1024 committed:
angular 15 update

feat(angular 15 update):
- Angular Material 15 migration.

pka1024 committed:
Merge branch 'hetida:release' into HDOS-675

pka1024 committed:
Merge branch 'HDOS-675' of https://github.com/pka1024/hetida-designer into HDOS-675

pka1024 committed:
angular 15 update

feat(angular 15 update):
- Angular Material 15 migration.
- Context-menu fix.

pka1024 committed:
angular 15 update

feat(angular 15 update):
- Updated eslint.

pka1024 committed:
angular 15 update

feat(angular 15 update):
- Updated to node version 16.20.2.
- Updated prettier.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- Updated typescript.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- ng update @angular/core@16 @angular/cli@16 --force.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- ng update @angular/material@16 --force.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- ng update @ngrx/store@16.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- npm i angular-auth-oidc-client@16.0.1.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- npm i @danielmoncada/angular-datetime-picker@16.1.0.
- npm i ngx-monaco-editor-v2@16.0.1.
- npm i monaco-editor@0.41.0.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- npm i @angular-eslint/schematics@16.3.1.
- npm i @angular-eslint/builder@16.3.1.
- npm i @angular-eslint/eslint-plugin@16.3.1.
- npm i @angular-eslint/eslint-plugin-template@16.3.1.
- npm i @angular-eslint/template-parser@16.3.1.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- npm i @typescript-eslint/eslint-plugin@7.7.1".
- npm i @typescript-eslint/parser@7.7.1.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- updated node to version "18.20.2".
- updated eslint to version "8.57.0".
- updated angular-plotly.js to version "5.2.2".
- updated @types/plotly.js to version "2.29.4".

pka1024 committed:
angular 16 update

feat(angular 16 update):
- auth guard migration.
- removed old comment.
- disabled "auth not enabled" button if not enabled.
- angular material migration.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- updated eslint-plugin-jsdoc.

pka1024 committed:
Merge branch 'develop' into HDOS-690

pka1024 committed:
angular 16 update

feat(angular 16 update):
- Updated katex.
- Removed  "preserveSymlinks": true from unit tests config.
- Updated Hetida-Flowchart packages to version 0.7.0.

pka1024 committed:
Merge branch 'develop' into HDOS-690

pka1024 committed:
angular 16 update

feat(angular 16 update):
- added routing.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- added routing.
- fixed query parameter.
- fixed Unit Tests.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- minor review changes.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- updated GitHub action too use node v18.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- updated GitHub action too use node v18.20.2.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- e2e-tests migration.
- auth enabled fixed for the logout menu.

pka1024 committed:
angular 17 update

feat(angular 17 update):
- zone.js and typescript updated.

pka1024 committed:
angular 17 update

feat(angular 17 update):
- ng update @angular/core@17 @angular/cli@17 --force.

pka1024 committed:
angular 17 update

feat(angular 17 update):
- ng update @angular/material@17 --force.

stewit committed:
upgrade node version in shell.nix

pka1024 committed:
angular 17 update

feat(angular 17 update):
- ng update @ngrx/store@17.

pka1024 committed:
angular 17 update

feat(angular 17 update):
- npm i angular-auth-oidc-client@17.1.0.

pka1024 committed:
angular 17 update

feat(angular 17 update):
- npm i @danielmoncada/angular-datetime-picker@17.0.0.

pka1024 committed:
angular 17 update

feat(angular 17 update):
- npm i @danielmoncada/angular-datetime-picker-moment-adapter@4.0.0.

pka1024 committed:
angular 17 update

feat(angular 17 update):
- npm i ngx-monaco-editor-v2@17.0.1.
- npm i monaco-editor@0.44.0.

pka1024 committed:
angular 17 update

feat(angular 17 update):
- npm i @angular-eslint/builder@17.5.2.
- npm i @angular-eslint/eslint-plugin@17.5.2.
- npm i @angular-eslint/eslint-plugin-template@17.5.2.
- npm i @angular-eslint/schematics@17.5.2.
- npm i @angular-eslint/template-parser@17.5.2.

stewit committed:
Update nixpkgs version

Commit:
update github action versions

Commit:
fix node version

Commit:
more verbose playwright logging

Commit:
fix playwright target url

pka1024 committed:
angular 16 update

feat(angular 16 update):
- unit-test "http failure response" fixed.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- changed to node version 20.17.0.

Commit:
chore: bump github action versions

pka1024 committed:
angular 16 update

feat(angular 16 update):
- angular Material migration.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- angular Material migration.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- hetida-flowchart packages updated.

pka1024 committed:
Merge branch 'HDOS-690' into HDOS-698

pka1024 committed:
angular 17 update

feat(angular 17 update):
- angular Material migration.

pka1024 committed:
angular 17 update

feat(angular 17 update):
- updated jasmin, karma, playwright.
- removed the deprecated RouterTestingModule and imported the RouterModule.

pka1024 committed:
Merge branch 'develop' into HDOS-698

pka1024 committed:
angular 17 update

feat(angular 17 update):
- angular Material migration.

pka1024 committed:
angular 17 update

feat(angular 17 update):
- angular Material migration.
- removed jasminewd2 package.
- fixed a not disabled input field, in a published workflow.

pka1024 committed:
angular 17 update

feat(angular 17 update):
- hetida-flowchart packages updated.

pka1024 committed:
angular 18 update

feat(angular 18 update):
- updated typescript.

pka1024 committed:
angular 18 update

feat(angular 18 update):
ng update @angular/core@18 @angular/cli@18 --force

pka1024 committed:
angular 18 update

feat(angular 18 update):
- ng update @angular/material@18 --force.

pka1024 committed:
angular 18 update

feat(angular 18 update):
- updated async in unit-test.

pka1024 committed:
angular 18 update

feat(angular 18 update):
- ng update @ngrx/store@18.

pka1024 committed:
angular 18 update

feat(angular 18 update):
- npm i --force angular-auth-oidc-client@18.0.2.

pka1024 committed:
angular 18 update

feat(angular 18 update):
- npm i --force @danielmoncada/angular-datetime-picker@18.1.0.

pka1024 committed:
angular 18 update

feat(angular 18 update):
- npm i @danielmoncada/angular-datetime-picker-moment-adapter@5.0.0.

pka1024 committed:
angular 18 update

feat(angular 18 update):
- npm i ngx-monaco-editor-v2@18.1.0.
- npm i monaco-editor@0.50.0.

pka1024 committed:
angular 18 update

feat(angular 18 update):
- npm i @angular-eslint/builder@18.4.2.
- npm i @angular-eslint/eslint-plugin@18.4.2.
- npm i @angular-eslint/eslint-plugin-template@18.4.2.
- npm i @angular-eslint/schematics@18.4.2.
- npm i @angular-eslint/template-parser@18.4.2.

pka1024 committed:
angular 18 update

feat(angular 18 update):
- npm audit --fix

pka1024 committed:
angular 18 update

feat(angular 18 update):
- ng update @angular/core@18 @angular/cli@18.

pka1024 committed:
angular 18 update

feat(angular 18 update):
- npm update.

pka1024 committed:
angular 18 update

feat(angular 18 update):
- angular material migration.
- baseHref updated.
- tslib updated.
- @types/jasmine updated.
- @types/node updated.

pka1024 committed:
angular 18 update

feat(angular 18 update):
- updated packages.

pka1024 committed:
angular 18 update

feat(angular 18 update):
- angular 18 control flow migration.

pka1024 committed:
angular 18 update

feat(angular 18 update):
- updated packages.
- fixed e2e tests.
- control flow migration changes.

pka1024 committed:
Merge branch 'develop' into HDOS-703

pka1024 committed:
angular 18 update

feat(angular 18 update):
- control flow migration changes.

pka1024 committed:
angular 18 update

feat(angular 18 update):
- nodejs version changed.

pka1024 committed:
angular 18 update

feat(angular 18 update):
- nodejs version changed to the latest LTS v22.13.0.
- updated hetida-flowchart packages.
- fixed a subscription bug in the toolbar.
- fixed two timers to unsubscribe on closed components and not run further in the background.
- removed "compile-libs": "ngcc" from package.json, it's no longer supported since Angular 16.

pka1024 committed:
Merge remote-tracking branch 'upstream/develop' into HDOS-703

pka1024 committed:
Merge remote-tracking branch 'upstream/develop' into HDOS-703

thatsmeta commented on frontend/playwright/test-cases/fixtures/fixture.ts:
```diff
@@ -9,11 +9,7 @@ type HetidaDesignerFixture = {
-        '[ERROR] 0-undefined - The clientId is required and missing from your config!',
-        '[ERROR] 0-undefined - Validation of config rejected with errors. Config is NOT set.'
-      ];
+      const ignoreErrorMessages = ['Error: Canceled: Canceled'];
```
// this is an error thrown by the monaco editor during e2e testing, we can safely ignore it

thatsmeta commented on frontend/src/app/components/documentation-editor-dialog/documentation-editor.component.html:
```diff
@@ -7,25 +7,26 @@
-      <ng-container *ngIf="editMode">Save</ng-container>
+      @if (!editMode) {
+        Edit
+      }
```
else

thatsmeta commented on frontend/src/app/components/popover-transformation/popover-transformation.component.html:
```diff
@@ -1,78 +1,77 @@
+        >
+      </div>
+    }
+    @if (showPopoverData.showPreview) {
```
else

thatsmeta commented on frontend/src/app/components/protocol-viewer/protocol-viewer.component.html:
```diff
@@ -1,17 +1,16 @@
-      </ng-container>
-      <ng-container *ngIf="!isExpanded">
+      }
+      @if (!isExpanded) {
```
else

thatsmeta commented on frontend/src/app/components/toolbar/toolbar.component.html:
```diff
@@ -23,29 +23,34 @@
-      *ngIf="isWorkflow"
-      >zoom_out</mat-icon
-    >
+    @if (isWorkflow) {
```
alles in eine klammer

thatsmeta commented on frontend/src/app/components/transformation-context-menu/transformation-context-menu.component.html:
```diff
@@ -11,27 +11,27 @@
-    <button mat-menu-item *ngIf="_isNotPublished" (click)="delete()">
-      Delete...
-    </button>
+    @if (_isNotPublished) {
```
in eine klammer

pka1024 committed:
angular 18 update

feat(angular 18 update):
- review changes.

pka1024 committed:
angular 18 update

feat(angular 18 update):
- updated nodejs version to LTS v22.13.0.
- removed deprecated compile-libs used by the old view engine in Angular v15 and before.
